src/common/neon_sse.h

```diff
@@ -0,0 +1,84 @@
+#ifndef NEON_SSE_H
+#define NEON_SSE_H
+
+#include <arm_neon.h>
+#include <stdlib.h>
+
+typedef int32x4_t	__m128i;
+typedef float32x4_t	__m128;
+
+static __inline void *
+__mm_malloc(size_t size, size_t alignment)
+{
+	void *ptr;
+
+	if (posix_memalign(&ptr, alignment, size) == 0)
+		return ptr;
+	else
+		return NULL;
+}
+
+static __inline void
+__mm_free(void *ptr)
+{
+	free(ptr);
+}
+
+static __inline void
+_mm_clflush(void const *p __attribute__((unused)))
+{
+	/* Cannot use __builtin__clear_cache */
+	/* Currenly not implemented */
+}
+
+static __inline void
+_mm_sfence(void)
+{
+	__atomic_thread_fence(__ATOMIC_SEQ_CST);
+}
+
+static __inline __m128i
+_mm_loadu_si128(__m128i *p)
+{
+	return (__m128i)(vld1q_s16((int16_t *)(p)));
+}
+
+static inline void
+_mm_stream_si128(__m128i *p, __m128i a)
+{
+	__builtin_memcpy(p, &a, sizeof(a));
```

Comment:
This is also misleading.
the `_mm_stream_si128` is expected to map to the `movntdq` instructions, and our code relies using this for non-temporal stores.
Does __builtin_memcpy guarantee that?
Unless we find an equivalent, we should just say that:
128bit streaming stores are not available.